AGENTS.md

```diff
@@ -0,0 +1,140 @@
+  • iOS targets: Follow existing deployment targets in package/podspec files; don’t lower without approval
+  • CI: GitHub Actions (assume PR validation on build + tests + lint)
+  • Linters & docs: SwiftLint (via Mint), Vale for prose (run phase exists), Jazzy/DocC where applicable
+
+### Project layout (high level)
+
+Sources/
+  StreamChat/            # Core chat client, models, networking, state
+  StreamChatUI/          # UIKit components
+Tests/
+  StreamChatTests/
+  StreamChatUITests/
+
+Use the closest folder’s patterns and conventions when editing.
+
+### Local setup (SPM)
+  1.  Open the repository root in Xcode (Package.swift is present), resolve packages.
+  2.  Select the intended scheme (see Schemes below), pick an iOS Simulator (e.g., iPhone 15), then Build.
+
+### Alternative: CocoaPods consumer check
+
+If you need to validate as a Pod consumer:
+  1.  Create a throwaway sample app.
+  2.  Add to Podfile:
+
+```
+use_frameworks!
+pod 'StreamChatUI', '~> 4.0'
+# core only: pod 'StreamChat', '~> 4.0'
+```
+
+  3.  pod install then build the sample.
+
+### Schemes
+
+Common scheme names in this repo include (exact names may evolve):
+  • StreamChat
+  • StreamChatUI
+  • Corresponding …Tests schemes
+
+Agents should query existing schemes from the project before invoking xcodebuild.
+
+### Build & test commands (CLI)
+
+Prefer Xcode for day-to-day work; use CLI for CI parity & agent automation.
+
+Swift Package (workspace-based) build:
+
+```
+xcodebuild \
+  -scheme StreamChat \
+  -destination 'platform=iOS Simulator,name=iPhone 15' \
+  -configuration Debug build
+```
+
+Run unit tests for all modules:
+
+```
+xcodebuild \
+  -scheme StreamChat \
+  -destination 'platform=iOS Simulator,name=iPhone 15' \
+  -configuration Debug test
+```
+
+```
+xcodebuild \
+  -scheme StreamChatUI \
+  -destination 'platform=iOS Simulator,name=iPhone 15' \
+  -configuration Debug test
+```
+
+If a Makefile is present, prefer the provided targets (e.g., make build, make test, make format, make lint). Always run make help to discover tasks.
+
+Linting & formatting
+  • SwiftLint is part of onboarding and may be installed via Mint. Run before committing:
+
+```
+swiftlint --strict
+```
+
+  • Respect .swiftlint.yml and any repo-specific rules. Do not suppress rules broadly; justify and scope exceptions.
+
+Commit / PR conventions
+  • Update CHANGELOG entries for user-visible SDK changes.
+  • Keep PRs small and focused; include tests.
+  • Follow the project’s “zero warnings” policy—fix new warnings and avoid introducing any.
+  • For UI changes, attach comparison screenshots (before/after) where feasible.
+  • Ensure public API changes include docs and migration notes.
+
+Testing policy
+  • Add/extend tests in the matching module’s Tests/ folder.
+  • Cover:
+  • Core models & API surface (StreamChat)
+  • View Controllers and UI behaviors (StreamChatUI)
+  • Use fakes/mocks from the test helpers provided by the repo when possible.
```

Comment:
Should not the bullet points after `Cover:` be with additional tabulation?

Reply:
done, good catch